course3/homework-string/my_string.h

```diff
@@ -1,6 +1,63 @@
 #pragma once
+#include <memory>
+#include <iostream>
+#include <cstring>
 
 class MyString
 {
-	// TODO
+private:
+    std::unique_ptr<char[]> data;
+public:
+	MyString();  			// default constructor
+	MyString(const char*);		// string literal constructor
+	MyString(const MyString&);	// copy constructor	
+
+	bool empty() const;
+	int size() const;
+
+	friend bool operator== (const MyString &s1, const MyString &s2); // equality operator is friended
+	friend bool operator!=  (const MyString &s1, const MyString &s2);
+	char& operator[](const int); // indexing operator
+	char& operator[](const int) const; // const indexing operator
+};
+
+MyString::MyString():MyString(""){};
+
+MyString::MyString(const char* inchar){
+	int n = std::strlen(inchar);
+	data = std::unique_ptr<char[]>(new char[n]);
+ 	std::strcpy(data.get(), inchar);
+};
+
+MyString::MyString(const MyString& that): data(new char[that.size()]){
+	std::strcpy(data.get(), that.data.get());
+}
+
+
+bool MyString::empty() const{
+	return size() == 0;
+};
+
+int MyString::size() const{
+	return std::strlen(data.get());
 };
+
+bool operator==(const MyString &s1, const MyString &s2){
+//	std::cout << "comparison operator " << s1.data.get()<< "==" << s2.data.get() << "?" << std::endl;
+	return std::strcmp(s1.data.get(), s2.data.get()) == 0;
+}
+
+bool operator!=(const MyString &s1, const MyString &s2){
+	return !(s1==s2);
+}
+
+char& MyString::operator[](const int i){
+	return data.get()[i];
+}
+
+char& MyString::operator[](const int i) const{
+	char a = data.get()[i];
+	char &ref_a = a;
+	return ref_a;
```

Comment:
whoa! You are returning a pointer (reference) to local variable. The variable is no longer valid when the function returns.

I know why you did it; it has to do with const-correctness. Please, compare the signature of your operator[] with the ones of ```std::string```